contracts/Registrar.sol

```diff
@@ -26,25 +26,46 @@ import "./OpsManaged.sol";
 import "./OpenSTValueInterface.sol";
 import "./OpenSTUtilityInterface.sol";
 
-/// @title Registrar - registers for utility tokens
+/**
+ *  @title Registrar contract which implements OpsManaged.
+ *
+ *  @notice Contains functions that register for utility tokens.
+ */
 contract Registrar is OpsManaged {
+	
+	/** Storage */
 
-    /*
-     *  Storage
-     */
-    // mapping(uint256 /* chainId */ => CoreInterface) cores;
+	// mapping(uint256 /* chainId */ => CoreInterface) cores;
 
-    /*
-     *  Public functions
-     */
-    constructor() public
-        OpsManaged()
-    {
-    }
+	/**  Public functions */
+
+	/**
+	 *  @notice Contract Constructor.
+	 */
+
+	 constructor() public
+	 	OpsManaged()
+	 	{ }
 
-    /*
-     *  OpenSTValue
-     */
+
+	/**
+	 *  @notice External function confirmRedemptionIntent.
+	 *
+	 *  @dev Only callable by Ops.
+	 *
+	 *  @param _uuid Uuid of the token.
+	 *  @param _redeemer Address of the redeemer.
+	 *  @param _redeemerNonce Nonce of the redeemer.
+	 *  @param _beneficiary Address of beneficiary on the value chain.
+	 *  @param _amountUT Amount of utility tokens to redeem.
+	 *  @param _redemptionUnlockHeight Block height upto which redemption is locked.
+	 *  @param _hashLock Hash lock for the redeem request.
+	 *  @param _blockHeight Current block height.
```

Comment:
I don't believe "current" is correct (because this block height is the value included in the redemption intent hash which comes from the utility chain).

Reply:
- block height at which redemption intent exists?

Reply:
Going with block height when hashed.